### PR TITLE
Adding `LargeProfileSummaryView`

### LINKED ISSUE #184: Create `LargeProfileSummaryView`

### PR DESCRIPTION
**Pull request by etoledom**
Closes #184 

### Description

Implementing `LargeProfileSummaryView`.

**Note**: This is a simple copy/paste of the `LargeProfileView`, and changing the layout as needed. I tested a different approach later on to allow sharing this common code between components. This PR is not intended to try this code sharing approach.

Design | Implementation

### CODE REVIEW

File: Sources/GravatarUI/ProfileFields/PersonalInfoField.swift [hunk 1]
```diff
@@ -37,4 +39,10 @@ public struct PersonalInfoBuilder {
         label.textColor = paletteType.palette.foreground.secondary
         return self
     }
+
+    @discardableResult
+    public func alignment(_ alignment: NSTextAlignment) -> PersonalInfoBuilder {
+        label.textAlignment = alignment
+        return self
+    }
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileFields/PersonalInfoField.swift:47]: I think it's better if we don't provide "public" methods for things that are NOT Gravatar related in these builders. UILabel has many other properties like this... We can easily end up with many public methods... And the more public types we provide more things to worry when changing something. So how about we keep this one internal, WDYT?

File: Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift [hunk 2]
```diff
@@ -0,0 +1,120 @@
+import Gravatar
+import UIKit
+
+public protocol ProfileCardSummaryModel: DisplayNameModel, PersonalInfoModel, AvatarIdentifierProvider {}
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift:4]: I wonder if we should drop "card" from these model names as well? 🤔 Like:

> ProfileSummaryModel

And the other one that LargeProfileView uses:

> ProfileModel

WDYT?

File: Demo/Demo/Gravatar-Demo/DemoLargeProfileViewController.swift [hunk 3]
```diff
@@ -32,10 +32,16 @@ class DemoLargeProfileViewController: UIViewController {
...
     }()
-    
+
+    lazy var largeProfileSummaryView: LargeProfileSummaryView = {
+        let view = LargeProfileSummaryView(frame: .zero, paletteType: preferredPaletteType)
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.avatarImageView.gravatar.activityIndicatorType = .activity
+        return view
+    }()
+//
```
Comment (pinarol) [inline Demo/Demo/Gravatar-Demo/DemoLargeProfileViewController.swift:42]: `//` can be removed

File: Demo/Demo/Gravatar-Demo/DemoLargeProfileViewController.swift [hunk 4]
```diff
@@ -44,7 +50,7 @@ class DemoLargeProfileViewController: UIViewController {
 
         return stack
     }()
-
+//
```
Comment (pinarol) [inline Demo/Demo/Gravatar-Demo/DemoLargeProfileViewController.swift:53]: `//` can be removed

File: Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    public func update(with model: ProfileCardSummaryModel) {
+        Configure(displayNameLabel).asDisplayName().content(model).palette(paletteType)
+        Configure(personalInfoLabel).asPersonalInfo().content(model).palette(paletteType).alignment(.center)
+    }
+
+    public func loadAvatar(
+        with avatarIdentifier: AvatarIdentifier,
+        placeholder: UIImage? = nil,
+        rating: Rating? = nil,
+        preferredSize: CGSize? = nil,
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift:86]: Again I notice this here but it's same in `LargeProfileView`. I think we should remove `preferredSize` parameter from this method. The caller party don't control the size of the UImageView so they don't have a reason to pass a `preferredSize`.

File: Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+        self.paletteType = .system
+        super.init(frame: frame)
+        addSubview(rootStackView)
+        layoutMargins = UIEdgeInsets(top: .margin, left: .margin, bottom: .margin, right: .margin)
+        NSLayoutConstraint.activate([
+            layoutMarginsGuide.topAnchor.constraint(equalTo: rootStackView.topAnchor),
+            layoutMarginsGuide.leadingAnchor.constraint(equalTo: rootStackView.leadingAnchor),
+            layoutMarginsGuide.trailingAnchor.constraint(equalTo: rootStackView.trailingAnchor),
+            layoutMarginsGuide.bottomAnchor.constraint(equalTo: rootStackView.bottomAnchor),
+        ])
```
Comment (pinarol) [inline Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift:63]: I know this is taken from `LargeProfileView` but I think we should also add `refresh(with: paletteType)` here like we do in the convenience init, just for sake of consistency... 

Same for `LargeProfileView`.

**Review by pinarol — COMMENTED** [review]
Looks good. 🎉  I dropped some small comments, some of them also apply to `LargeProfileView`. We can talk about those quickly and then merge this.

**Review by pinarol — APPROVED** [review]

File: Sources/GravatarUI/ProfileFields/PersonalInfoField.swift (hunk 1 above)
Comment (etoledom) [reply Sources/GravatarUI/ProfileFields/PersonalInfoField.swift:47]: Will do on a next PR 👍

File: Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift (hunk 2 above)
Comment (etoledom) [reply Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift:4]: Will do on a next PR 👍

File: Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift (hunk 5 above)
Comment (etoledom) [reply Sources/GravatarUI/ProfileView/LargeProfileSummaryView.swift:86]: Will do on a next PR 👍

**Pull request merged**